sdk/data/azcosmos/cosmos_container_read_many.go

```diff
@@ -4,19 +4,252 @@
 package azcosmos
 
 import (
+	"bytes"
 	"context"
 	"errors"
 	"runtime"
 	"sync"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	azlog "github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
 	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos/queryengine"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/log"
 )
 
-// executeReadManyWithEngine executes a query using the provided query engine.
+// Executes a query using the provided query engine.
 func (c *ContainerClient) executeReadManyWithEngine(queryEngine queryengine.QueryEngine, items []ItemIdentity, readManyOptions *ReadManyOptions, operationContext pipelineRequestOptions, ctx context.Context) (ReadManyItemsResponse, error) {
```

Comment:
This is looking OK to me. I'd like to try and unify the pipeline "driving" code between read many and cross-partition query, since it has all the same basic patterns. The concurrency you're doing here should _also_ be possible with the query pipeline once we make some improvements. But I'm OK with doing that refactoring later, once we have an end-to-end working ReadMany implementation